Example project not compiling properly

# Steps to reproduce

```sh
$ git clone https://github.com/Microsoft/TypeScript-React-Native-Starter
$ git checkout 3fc367a1ab877316ed2dd3001e113ddc2080930e
$ yarn install
$ tsc -p tsconfig.json --outDir build
```

# Result

```
node_modules/@types/react-native/index.d.ts(3415,42): error TS2304: Cannot find name 'Map'.
node_modules/@types/react-native/index.d.ts(3428,42): error TS2304: Cannot find name 'Map'.
node_modules/@types/react-native/index.d.ts(8685,18): error TS2717: Subsequent property declarations must have the same type.  Property 'geolocation' must be of type 'Geolocation', but here has type 'GeolocationStatic'.
```

# Expected Result

To compile properly.

--- a/ExampleProject/tsconfig.json
+++ b/ExampleProject/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     /* Basic Options */
-    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
     "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
     // "lib": [],                             /* Specify library files to be included in the compilation. */
     // "allowJs": true,                       /* Allow javascript files to be compiled. */

